File: pset6/sentimental/vigenere/vigenere.py
import sys

# check whether only a single argument was given
if len(sys.argv) == 2 and sys.argv[1].isalpha():
    # set key variable
    key = sys.argv[1]
    # get and set key_len for better performance
    key_len = len(key)

    # get plaintext string from user
    plaintext = str(input("plaintext: "))

    # initiate enciphering
    print('ciphertext: ', end='')
    # initialize counter
    k_counter = 0
    for c in range(len(plaintext)):
        # encipher alphabetical characters
        if plaintext[c].isalpha():

            # set variable for string indices
            index = k_counter % key_len

            # handle upper and lowercases
            if key[index].isupper():
                k = ord(key[index]) - 65
            else:
                k = ord(key[index]) - 97

            # add enciphered characters to plaintext variable
            if plaintext[c].isupper():
                char = chr((ord(plaintext[c]) - 65 + k) % 26 + 65)
            else:
                char = chr((ord(plaintext[c]) - 97 + k) % 26 + 97)
        # add non-alphabetical without enciphering
        else:
            char = plaintext[c]

        # increment counter
        k_counter += 1

        # print enciphered character
        print(char, end='')
    # new line
    print()

# exit if command line argument is invalid
else:
    print("Usage: python {} KEYWORD".format(sys.argv[0]))
    print("\nPLEASE NOTE:\nKEYWORD should only contain alphabetical characters and no numbers")
    exit(1)

File: pset2/vigenere/vigenere.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
// get keyword from command line input
int main(int argc, string argv[])
{
    if (argc == 2)
    {
        string key = argv[1];
        // check if all chars in keyword are alphabetical chars
        // exit if not
        for (int c = 0; c < strlen(key); c++)
        {
            if (isalpha(key[c]) == false)
            {
                // exit if char other than alpha chars are present
                printf("Usage: %s KEYWORD\n", argv[0]);
                exit(1);
            }
        }
        // get user input (plaintext)
        string plaintext = get_string("plaintext: ");
        // set a counter for chars in key
        int k_counter = 0;
        printf("ciphertext: ");
        for (int c = 0; c < strlen(plaintext); c++)
        {
            // check if char isalpha
            if (isalpha(plaintext[c]))
            {
                // calc k
                int k = tolower(key[k_counter % strlen(key)]) - 97;
                // handle uppercase and lowercase and print char
                if (isupper(plaintext[c]))
                {
                    printf("%c", ((plaintext[c] - 65) + k) % 26 + 65);

                }
                else if (islower(plaintext[c]))
                {
                    printf("%c", ((plaintext[c] - 97) + k) % 26 + 97);
                }
                k_counter++;
            }
            else
            {
                // if char is not alphabetical char, print plaintext char
                printf("%c", plaintext[c]);
            }
        }
        // print new line
        printf("\n");

    }
    else
    {
        // print error message and exit
        printf("Usage: %s KEYWORD\n", argv[0]);
        exit(1);
    }
}
